akka-typed/src/main/java/akka/typed/javadsl/Actor.java

```diff
@@ -0,0 +1,377 @@
+      return Same();
+    }
+
+    @Override
+    public Behavior<T> message(akka.typed.ActorContext<T> ctx, T msg) throws Exception {
+      message.apply(ctx, msg);
+      return Same();
+    }
+  }
+
+  private static class Tap<T> extends Behavior<T> {
+    final Procedure2<ActorContext<T>, Signal> signal;
+    final Procedure2<ActorContext<T>, T> message;
+    final Behavior<T> behavior;
+
+    public Tap(Procedure2<ActorContext<T>, Signal> signal, Procedure2<ActorContext<T>, T> message,
+        Behavior<T> behavior) {
+      this.signal = signal;
+      this.message = message;
+      this.behavior = behavior;
+    }
+
+    private Behavior<T> canonicalize(Behavior<T> behv) {
+      if (Behavior.isUnhandled(behv))
+        return Unhandled();
+      else if (behv == Same())
+        return Same();
+      else if (Behavior.isAlive(behv))
+        return new Tap<T>(signal, message, behv);
+      else
+        return Stopped();
+    }
+
+    @Override
+    public Behavior<T> management(akka.typed.ActorContext<T> ctx, Signal msg) throws Exception {
+      signal.apply(ctx, msg);
+      return canonicalize(behavior.management(ctx, msg));
+    }
+
+    @Override
+    public Behavior<T> message(akka.typed.ActorContext<T> ctx, T msg) throws Exception {
+      message.apply(ctx, msg);
+      return canonicalize(behavior.message(ctx, msg));
+    }
+  }
+
+  /**
+   * Mode selector for the `restarter()` wrapper that decides whether an actor
+   * upon a failure shall be restarted (to clean initial state) or resumed (keep
+   * on running, with potentially compromised state).
+   * 
+   * Resuming is less safe. If you use `OnFailure.RESUME` you should at least
+   * not hold mutable data fields or collections within the actor as those might
+   * be in an inconsistent state (the exception might have interrupted normal
+   * processing); avoiding mutable state is possible by returning a fresh
+   * behavior with the new state after every message.
+   */
+  public enum OnFailure {
+    RESUME, RESTART;
+  }
+
+  private static Function2<Object, Object, Object> _unhandledFun = (ctx, msg) -> Unhandled();
+
+  @SuppressWarnings("unchecked")
+  private static <T> Function2<ActorContext<T>, Signal, Behavior<T>> unhandledFun() {
+    return (Function2<ActorContext<T>, Signal, Behavior<T>>) (Object) _unhandledFun;
+  }
+
+  private static Procedure2<Object, Object> _doNothing = (ctx, msg) -> {
+  };
+
+  @SuppressWarnings("unchecked")
+  private static <T> Procedure2<ActorContext<T>, Signal> doNothing() {
+    return (Procedure2<ActorContext<T>, Signal>) (Object) _doNothing;
+  }
+
+  /**
+   * Construct an actor behavior that can react both to lifecycle signals and
+   * incoming messages. After spawning this actor from another actor (or as the
+   * guardian of an [[akka.typed.ActorSystem]]) it will be executed within an
+   * [[ActorContext]] that allows access to the system, spawning and watching
```

Comment:
right, I have not yet checked doc generation at all—will make it work correctly, of course